hibernate-core/src/main/java/org/hibernate/engine/jdbc/env/internal/JdbcEnvironmentInitiator.java

```diff
@@ -239,6 +224,27 @@ else if ( useJdbcMetadata ) {
 				log.unableToObtainConnectionToQueryMetadata( e );
 			}
 		}
+		else {
+			if (someExplicitVersionSet) {
```

Comment:
I'd have to chew on this a bit; I think it's risky to introduce the condition based on `AvailableSettings.DIALECT` because until now this was only evaluated in the`DialectFactory`, not in the initiator (which is something that I still don't understand tbh, it makes the logic hard to follow and messy), so imo this could well have some unwanted side-effects.

Reply:
You can split up the predicates to make it more readable. Not sure about side effects. I mean this only is an issue if people disable JDBC metadata usage, and explicitly configure database information.
If people configure a version but don't configure a database name or dialect, then the current logic would just ignore the configured version. The change I propose would respect the explicitly configured version when a dialect or database name is also configured, so IMO this change is safe. Please explain what you think could cause unwanted side effects.